apps/01PigLatin.js

```diff
@@ -7,11 +7,80 @@ prompt.start();
 
 function pigLatin(word) {
 
-    // Your code here
+  // Concept Check
+  //
+  // Please explain why we are using String.prototype.toLowerCase() here.
+  // For more info:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase
+  //
+  // Your Answer:
+  // ANSWER: it prevents needing to check all variations of words that have both lower case and capital letters.  'a' vs 'A' - can get time consuming to go thru whole alphabet.
+  word  = word.toLowerCase();
 
+  // Concept Check
+  //
+  // What is the significance of setting vowelIndex to -1?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
+  //
+  // Your Answer:
+  // ANSWER: an array with an index of -1 does not exist so it is a safe starting point for checking the indexes
+  var vowelIndex = -1; // Set it to assume there are no vowels
+
+  if ( ( word.indexOf('a') > -1 && word.indexOf('a') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('a');
+  }
+
+  if ( ( word.indexOf('e') > -1 && word.indexOf('e') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('e');
+  }
+
+  if ( ( word.indexOf('i') > -1 && word.indexOf('i') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('i');
+  }
+
+  if ( ( word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('o');
+  }
+
+  if ( ( word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('u');
+  }
+
+  if ( ( word.indexOf('y') > -1 && word.indexOf('y') < vowelIndex ) || vowelIndex === -1 ) {
+      vowelIndex = word.indexOf('y');
+  }
+
+  // Concept Check
+  //
+  // Why are we using String.prototype.slice() here? What role does vowelIndex play?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice
+  //
+  // Your Answer:
+  //  ANSWER: slice tells javascript where to cut off the word.  We want it to slice off the first letter(s) before the first vowel.  The index of the first vowel tells us where the first vowel is.
+  var firstPart = word.slice(0, vowelIndex);
+  var restWord = word.slice(vowelIndex, word.length);
+
+  if (vowelIndex === 0) {
+    return word + 'yay';
+  }
+  else {
+    return restWord + firstPart + 'ay';
+  }
 }
 
 
+// Concept Check
+//
+// Make sure you run this app with node in the console before answering this question.
+// Also, read Lesson 1 in the textbook.
+// http://intermediate.austincodingacademy.com/_book/lesson_one.html
+//
+// What does prompt.get() do for us?
+//
+// Your Answer:
+// ANSWER:  it allows the user of the app to enter any word they want so it is not hardcoded into the code
 function getPrompt() {
     prompt.get(['word'], function (error, result) {
```

Comment:
Great job answering the questions, @nina2cool!

![toast](http://31.media.tumblr.com/14f06975f3c63b7b60c637e5707d2388/tumblr_n0wn6i68Jy1r83n9go1_500.gif)